workshops/util.py

```diff
@@ -551,3 +551,32 @@ def update_event_attendance_from_tasks(event):
 
 def universal_date_format(date):
     return '{:%Y-%m-%d}'.format(date)
+
+
+def get_members(earliest, latest):
+    '''Get everyone who is a member of the Software Carpentry Foundation.'''
+
+    member_badge = Badge.objects.get(name='member')
+    instructor_badge = Badge.objects.get(name='instructor')
+    instructor_role = Role.objects.get(name='instructor')
+
+    # Everyone who is an explicit member.
+    explicit = Person.objects.filter(badges__in=[member_badge]).distinct()
+
+    # Everyone who qualifies by having taught recently.
+    implicit = Person.objects.filter(
+        task__role=instructor_role,
+        badges__in=[instructor_badge],
+        task__event__start__gte=earliest,
+        task__event__start__lte=latest
+    ).distinct()
+
+    # Merge the two sets.
+    return explicit | implicit
+
+
+def default_membership_cutoff():
+    "Calculate a default cutoff dates for members finding with `get_members`."
+    earliest = datetime.date.today() - 2 * datetime.timedelta(days=365)
+    latest = datetime.date.today()
```

Comment:
With a cutoff setup for calendar years, I'm not sure how to handle `latest`.  Do we project out to the end of the current year, and see what events an instructor is planning to teach at?  Do we project out to a particular event (e.g. whenever we plan on holding elections)?  To figure out the business logic around `earliest` and `latest`, we might want to consider:

1. The current time, since AMY's knowledge of instructors at past events is presumably more reliable than AMY's knowlege of instructors at future events.
2. When we care about applying the measurement.  I may be interested in the current set of members, or maybe I'm preparing for an election on January 10th, and want an estimate at the member set then.  Or maybe I want the set of instructors who will still be members on January 10th if everybody stops teaching today.